In [3]:
import random 
import pandas as pd 
import numpy as np 
import seaborn as sns 
import matplotlib.pyplot as plt 


In [4]:

random.seed(42)

In [6]:


df = pd.read_csv("data.csv")
df.head(2)


id Gender  Age  Driving_License  Region_Code  Previously_Insured  \
0   1   Male   44                1         28.0                   0   
1   2   Male   76                1          3.0                   0   

  Vehicle_Age Vehicle_Damage  Annual_Premium  Policy_Sales_Channel  Vintage  \
0   > 2 Years            Yes         40454.0                  26.0      217   
1    1-2 Year             No         33536.0                  26.0      183   

   Response  
0         1  
1         0


<br>

**Vehicle insurance:** is a contract where you pay premiums to an insurer for financial protection against losses from accidents, theft, or damage to your car, covering your own repair costs, injuries, and legal liability for damage or harm you cause to others.

<br>


* **Gender**: Customer gender (Male / Female)
* **Age**: Customer’s age
* **Driving_License**:

  * `1` → Has driving license
  * `0` → Does not have driving license
* **Region_Code**: Encoded region information
* **Previously_Insured**:

  * `1` → Customer already has vehicle insurance
  * `0` → Not previously insured
* **Vehicle_Age**: Age category of the vehicle (`< 1 Year`, `1–2 Years`, `> 2 Years`)
* **Vehicle_Damage**:

  * `Yes` → Vehicle was damaged in the past
  * `No` → No past damage
* **Annual_Premium**: Yearly insurance premium amount
* **Policy_Sales_Channel**: Encoded channel used for selling the policy
* **Vintage**: Number of days the customer has been associated with the company
* **Response (Target Variable)**:

  * `1` → Customer is interested in the insurance
  * `0` → Customer is not interested

<br>
<br>

**Annual_Premium:** <br>
✔ Yes, 40454 is money you pay and usually don’t get back. <br>
✔ It is the cost of risk protection, not savings. <br>

<br>
